pallets/gated-marketplace/src/lib.rs

```diff
@@ -341,10 +362,38 @@ pub mod pallet {
 			fee: u32,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?; // origin will be market owner
-			let m = Marketplace { label, fee: Permill::from_percent(fee), creator: who.clone(), };
+			let m = Marketplace { label, fee: Permill::from_percent(fee), creator: who.clone() };
 			Self::do_create_marketplace(who, admin, m)
 		}
 
+		/// Block or Unblock a user from apllying to a marketplace.
+		///
+		/// Blocks or Unblocks a user from applying to a marketplace.
+		///
+		/// ### Parameters:
+		/// - `origin`: The admin of the marketplace.
+		/// - `marketplace_id`: The id of the marketplace to block/unblock the user.
+		/// - `user`: The id of the user to block/unblock.`
+		///
+		/// ### Considerations:
+		/// - Once a user is blocked, the user won't be able to join the marketplace until unblocked.
+		#[pallet::weight(Weight::from_ref_time(10_000) + T::DbWeight::get().writes(1))]
+		pub fn block_user(
+			origin: OriginFor<T>,
+			marketplace_id: MarketplaceId,
+			block_args: BlockUserArgs<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			match block_args {
+				BlockUserArgs::BlockUser(user) => {
+					return Self::do_block_user(who, marketplace_id, user);
```

Comment:
Remove the return & the semicolon in each matched case, is not necessary since rust knows that if we don't use semicolons, it's a return
`Self::do_block_user(who, marketplace_id, user)`